test/services/EnglishTextBuilderSpec.scala

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import base.{EnglishLanguage, BaseSpec}
+import models.ocelot._
+import models.ui.{Currency, CurrencyPoundsOnly, DateStandard, LabelRef, Link, Text, Txt, Words}
+
+class EnglishTextBuilderSpec extends BaseSpec {
+
+  trait Test extends ProcessJson with EnglishLanguage {
+    val words1 = Words("This is a ", true)
+    val words2 = Words(" followed by ")
+    val words3 = Words(" and nothing")
+    val link1EnWords = "A link"
+    val link1CyWords = "Welsh, A link"
+    val link2EnWords = "Another Link"
+    val link2CyWords = "Welsh, Another Link"
+
+    val link1 = Link("https://www.bbc.co.uk", link1EnWords, false)
+    val link2 = Link("https://www.gov.uk", link2EnWords, false)
+    val urlMap1: Map[String, String] = Map("start" -> "dummy-path/start", "3" -> "dummy-path", "5" -> "dummy-path/blah", "34" -> "dummy-path/next")
+
+    val txtWithLinks = Phrase(
+      Vector(
+        s"[bold:This is a ][link:${link1EnWords}:https://www.bbc.co.uk] followed by [link:${link2EnWords}:https://www.gov.uk] and nothing",
+        s"[bold:Welsh, This is a ][link:${link1CyWords}:https://www.bbc.co.uk] Welsh, followed by [link:${link2CyWords}:https://www.gov.uk] Welsh, and nothing"
+      )
+    )
+
+    val brokenLinkPhrase = Phrase(Vector("Hello [link:Blah Blah:htts://www.bbc.co.uk]", "Welsh, Hello [link:Blah Blah:htts://www.bbc.co.uk]"))
+    implicit val stanzaIdToUrlMap: Map[String, String] = Map()
+
+    val answerWithNoHint = Phrase("Yes", "Welsh, Yes")
+    val answerWithHint = Phrase("Yes[hint:You agree with the assertion]", "Welsh, Yes[hint:Welsh, You agree with the assertion]")
+
+    val answer = Text("Yes")
+    val hint = Text("You agree with the assertion")
+  }
+
+  "TextBuilder placeholder parsing" must {
+
+    "Convert label reference with default output format placeholders within phrase to LabelRef TextItems" in new Test {
```

Comment:
In a number of the tests below the English and Welsh text are identical. As such we cannot be certain that the TextBuilder class is processing the English text.

Reply:
updated